modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/liferay.js

```diff
@@ -196,7 +196,13 @@ Liferay = window.Liferay || {};
 			},
 			method: 'POST',
 		})
-			.then((response) => response.json())
+			.then((response) => {
+				if (!response.ok) {
+					ioConfig.error();
```

Comment:
I think we need to also do something else here, otherwise the callback will be called twice, one here and one in the complete function, right?

Reply:
The error might get thrown twice, yea 🤔 

What do you propose?

Reply:
Mmm, we can throw an error here to stop the promise flow, which seems kinda dirty

Other better option may be to move the check to the latest `then`, somethink like

```javascript

.then((response) => Promise.all([Promise.resolve(response), response.json()] )
.then(([response, content]) => { 
    if(response.ok) {
       ioConfig.complete(content);
    }
   else {
      ioConfig.error();
   }
})
.catch(ioConfig.error);
```

thoughts?

Reply:
I'll test it out. I was mainly following the [docs](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#checking_that_the_fetch_was_successful). Thanks for the clue ❤️

Reply:
Your solution invokes the `console.log('error 2')` from the invocation only once, whereas mine does it twice. So thank you!
```
Liferay.Service(
  '/group/get-group',
  {
    groupId: 123
  },
  function(response) {
    console.log('success 2: ', response)
  },
  function(error) {
    console.log('error 2')
  },
)
```

Reply:
Nice 👏